packages/lib-utils/src/ws/types.ts

```diff
@@ -0,0 +1,27 @@
+import type { AnyObject } from '@monorepo/common';
+
+type GenericHandler<T = unknown> = (data: T) => void;
+export type OpenHandler = GenericHandler<never>;
+export type CloseHandler = GenericHandler<CloseEvent>;
+export type ErrorHandler = GenericHandler<Event>;
+/**
+ * The WebSocket data can be returned in an object state or in the raw string response passed.
+ *
+ * This is configured through `jsonParse` options flag.
+ * @see WSOptions
+ */
+export type MessageDataType = AnyObject | string;
```

Comment:
See other comment here: https://github.com/openshift/dynamic-plugin-sdk/pull/38/files#r819000771. 

Not sure if we want to push these types to the root types folder?

Reply:
Not sure we need to push all types into the same folder. That seems like a bad way to stay organized as we grow 🤔 

Not sure why we have a `types` folder if not for common stuff.

Reply:
It was a convention we started with.

For example, types for code in `packages/lib-utils/src/k8s` would live in `packages/lib-utils/src/types/k8s.ts` or `packages/lib-utils/src/types/k8s-whatever.ts` etc.

It's OK to put types close to relevant code. Also, we can always change the structure as necessary.